spec/lib/jobs/install-os-spec.js

```diff
@@ -406,6 +406,61 @@ describe('Install OS Job', function () {
         });
     });
 
+    it('should fail job if task notification received with failure', function(done) {
+        subscribeNodeNotification.restore();
+        subscribeNodeNotification = sinon.stub(
+            InstallOsJob.prototype, '_subscribeNodeNotification', function(_nodeId, callback) {
+                callback({
+                    nodeId: _nodeId,
+                    nodeIp: '1.1.1.1',
+                    status: 'fail'
+                });
+            });
+
+        waterline.graphobjects.findOne = sinon.stub().resolves(graph);
+        job.run()
+        .then(function(){
+            done(new Error("Expected job to fail"));
```

Comment:
The error should have been rejected by `_subscribeNodeNotification`, so `"Expected job to fail"` error makes no sense though this kind of code exists everywhere in our repo.